src/reader.rs

```diff
@@ -41,13 +42,15 @@ impl<'a> Reader<'a> {
     /// Returns `true` if the reader is at the end of the input, and `false`
     /// otherwise.
     #[inline]
+    #[must_use]
     pub fn at_end(&self) -> bool {
         self.i == self.input.len()
     }
 
     /// Returns `true` if there is at least one more byte in the input and that
     /// byte is equal to `b`, and false otherwise.
     #[inline]
+    #[must_use]
```

Comment:
This one is the one that makes the most sense to me.

I think `#[must_use]` is most appropriate for functions where one could reasonably ignore the return value accidentally. For the rest of the functions you've added it to, it seems implausible that somebody would accidentally call the function and ignore the return value.

In some sense, if a Rust function returns a result, you probably shouldn't ignore it. There should be some lint stronger than `unused_must_use` that enforces that, without one needing to add `#[must_use]` to every function.

WDYT?